File: src/org/cerion/symcalc/expression/function/Hold.java
package org.cerion.symcalc.expression.function;

import org.cerion.symcalc.Environment;
import org.cerion.symcalc.expression.Expr;
import org.cerion.symcalc.expression.FunctionExpr;
import org.cerion.symcalc.expression.number.IntegerNum;

public class Hold extends FunctionExpr {

    public Hold(Expr... e) {
        super(FunctionType.HOLD,e);
    }

    @Override
    protected Expr evaluate() {
        return this;
    }

    @Override
    protected int getProperties() {
        return Properties.HOLD.value;
    }
}

File: symcalc/src/main/kotlin/function/core/Hold.kt
package org.cerion.symcalc.function.core

import org.cerion.symcalc.expression.Expr
import org.cerion.symcalc.function.FunctionExpr

class Hold(vararg e: Expr) : FunctionExpr(*e) {

    override val properties: Int
        get() = Properties.HoldAll.value

    override fun evaluate(): Expr {
        return this
    }

    override fun validate() {
    }
}
